test/promise/evil-promises.js

```diff
@@ -20,7 +29,7 @@ describe('Evil promises should not be able to break invariants', function () {
     };
 
     var calledAlready = false;
-    Promise.resolve(evilPromise).then(function (value) {
+    new Promise(function (r) { r(evilPromise); }).then(function (value) {
```

Comment:
`Promise.resolve` has this sorta-sketchy "is the argument of the same class as `this`"  shortcut, which is what the `new Promise` version is explicitly avoiding.